### DISCUSSION TITLE
League Message Board Data

### DISCUSSION THREAD

**Discussion by Acarlso** [cwendt94/espn-api#429] [ESPN API]
Is it possible to extract league message board data?

**Comment by cwendt94** [comment]
That should be possible, are you wanting like the messages from the recent activity tab on ESPN?

**Comment by Acarlso** [comment]
This may be a tall order, but I'm looking for the entire contents of the League Message Board.  

For context, my 16 team keeper league of record ran on ESPN from 2014-2019 when we switched to Sleeper.  After 3 years of not reactivating the league, it appears to be soft deleted or something on ESPN's end and we can see the league but can not access or reactivate it from ESPN.com.  This python package has already been an incredible blessing whereby I was able to pull ALL of our historical data which all of the league had already written off and given up hope of being able to retrieve.  

My league has a tradition where the lowest scorer every week, as a punishment and to discourage tanking, has to write what we call the "League Note" which is expected to be some sort of creative writing exercise, hard hitting league specific analysis, etc. that everyone gets to enjoy with a cup of coffee prior to kickoff every Sunday.  The process has been refined over the years and we now upload the League Note to a shared drive, but in the early years the League Note was published to ESPN's League Message Board.  

The ability to pull all Message Board data would resurrect years of truly gold banter and short stories that may as well have burned along with the Library of Alexandria as far as everyone is concerned.  You would be a hero.

**Reply by cwendt94** [reply]
Very cool idea, I might bring that over to my league! Always need to keep the members inline from tanking lol!

Let me look into it! I think they have an API to get that information but not 100% sure.

**Reply by cwendt94** [reply]
I created a PR that will get all of the data from the msg board https://github.com/cwendt94/espn-api/pull/434. Because my league did not use the message board that much I am not sure how much data to pull from the API response, so right now I am returning all the data for each message. I would love to get some feedback if you could try out that PR and see what data you need from the message board response.

If you want to get all message types from the board you can call the function without any parameters
```python3
league.message_board()
```
To filter messages you can pass a string list of the message types you want. Here are a couple I saw from my league, not sure if it is all of the types
```python3
# Messages types ACTIVITY_TRANSACTIONS, ACTIVITY_STATUS, EMAIL, NOTE, MSG_BOARD_MATCHUP, ACTIVITY_SETTINGS
# ACTIVITY_SCHEDULE, MSG_BOARD
league.message_board(msg_types=['EMAIL', 'MSG_BOARD'])
```

**Reply by Acarlso** [reply]
This is seriously amazing!  All of the content that I'm looking for appears to be in msg_types=['MSG_BOARD'].  The data is really sparse prior to 2018 though and I'm wondering if something changed prior to that year.  I called league.message_board() wide open for 2014-2017 and only returned a single record of msg_types=['NOTE'] from 2014 and another record of msg_types=['NOTE'] from 2016.  2015 and 2017 returned no message data at all.  Unfortunately I can't access the league UI in ESPN to confirm whether there is actually message data to return or not.  It's entirely possible that we did not use the message board prior to 2018, but no one can remember.

**Reply by cwendt94** [reply]
Interesting, in 2018 thats when they updated their API to v2 and removed a lot of the old data from previous years. I wonder if they removed the message board data.
